Using Exchange topic for the Connector Overview set of topics (#528)
mulesoft/docs-connectors: modules/ROOT/pages/introduction/use-exchange.adoc

> = Use Exchange to Discover Connectors, Templates, and Examples

Reviewer: Instructions for edit:
I have a new topic for you to edit when you have a chance. This is a topic from the Introduction to Connectors guide (though we may publish it on its own before the other topics are finished): https://github.com/mulesoft/docs-connectors/pull/528/. This is an initial edit since the topic is new. (edited)

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../../assets/images/

Use Anypoint Exchange (Exchange) to discover connectors, templates, and examples, and to incorporate them into your app. Exchange serves as a repository for connectors, templates, and examples provided both by MuleSoft and by other organizations. Searches in Exchange apply only to the Exchange organization you are viewing.

> Search in the https://www.mulesoft.com/exchange/[Public Exchange] to access MuleSoft's public assets. If you search in a private organization's Exchange, you need access rights to that organization.

Reviewer: Should we specify "a private organization's implementation of Exchange"?

For more information about searching in Exchange, see xref:exchange::to-find-info.adoc[Find Asset Information].

== Discover and Add Connectors to Flow Designer and Studio Projects

> Discover connectors by starting the search in Exchange or by accessing Exchange from within an Anypoint Flow Designer or Anypoint Studio project:

Reviewer: ```suggestion
Discover connectors either by starting the search in Exchange or by accessing Exchange from within an Anypoint Flow Designer or Anypoint Studio project:
```

> * Start in Exchange to view a list of all, or a subset of MuleSoft-provided connectors. From there, you can view a connector in Flow Designer or download it to Studio.

Reviewer: ```suggestion
* Start in Exchange to view a list of all or a subset of MuleSoft-provided connectors. 
From there, you can view a connector in Flow Designer or download it to Studio.
```

* Access Exchange from within a Flow Designer or Studio project to locate a template applicable to your use case.

> === Discover all Connectors Provided by MuleSoft

Reviewer: ```suggestion
=== Discover All Connectors Provided by MuleSoft
```

To use Exchange as a starting point for discovering all or a subset of MuleSoft-provided connectors:

. https://www.mulesoft.com/exchange/[Access Exchange].
. Under *All assets* on the left side of the screen, click *Provided by MuleSoft*.
. Select *Connectors* in the drop-down list to the left of the search bar.

> . To filter the list of connectors, add text to the search field. For example, if you enter `Salesforce`, the search returns all Salesforce connectors.

Reviewer: ```suggestion
. To filter the list of connectors, add text to the search field. 
For example, if you enter `Salesforce`, the search returns all Salesforce connectors.
```

> . Click on a connector to view its details.

Reviewer: ```suggestion
. Click a connector to view its details.
```

[[add-connector-to-flow-designer]]
=== Add a Connector to a Flow Designer App

> When you are working in a Flow Designer project, you can add a connector directly to the project. You can only use Flow Designer for supported connectors.

Reviewer: ```suggestion
When you are working in a Flow Designer project, you can add a connector directly to the project. You can use Flow Designer only for supported connectors.
```

. Log into Anypoint Platform.
. Open Design Center and create a new Mule app in Flow Designer.
. In the *Let's Get Started wizard*, click *Go straight to canvas*, in the bottom-left corner.

> . Click the on Trigger card.

Reviewer: ```suggestion
. Click on the Trigger card.
```

. Select the connector.
. Continue designing the flow, as described in the User Guide for your connector.

[[import-connector-studio]]
=== Import a Connector into a Studio Project

To import a connector into a Studio project:

> . From your Studio project, click the Exchange icon *(X)* in the upper-left of the Studio task bar.

Reviewer: ```suggestion
. From your Studio project, click the Exchange icon *(X)* in the upper-left of the Studio taskbar.
```

. Log into Exchange using your Anypoint Platform username and password.
. Search for the connector.
. Select the connector and click *Add to project*.
. Follow the prompts to add the connector.

=== Next Step

Follow the directions in the connector's User Guide for configuring the connector in Design Center (using Flow Designer) or Studio.

== Discover and Add Templates to Flow Designer or Studio

A template is a pre-built project that you can use as a starting point for applicable use cases. Many templates include one or more connector configurations.

Discover templates by starting in Exchange or by accessing Exchange from within Flow Designer or Studio:

> * Start in Exchange to view a list of all, or a subset of MuleSoft-provided templates. From there, you can view a template in Flow Designer or download it to Studio.

Reviewer: ```suggestion
* Start in Exchange to view a list of all or a subset of MuleSoft-provided templates. 
From there, you can view a template in Flow Designer or download it to Studio.

```

* Access Exchange from within Flow Designer or Studio to locate a template that fits your use case.

When you locate a template in Exchange, the landing page of the associated connector contains:

* An overview of what the template does
* Prerequisites for using the template
* Instructions for configuring the template

> === Discover all MuleSoft Templates

Reviewer: ```suggestion
=== Discover All MuleSoft Templates
```
It's not always the length that determines title caps.; it's also the part of speech.

You can use Exchange as a starting point for discovering MuleSoft-provided templates:

. https://www.mulesoft.com/exchange/[Log into Exchange].
. Under *All assets* on the left side of the screen, click *Provided by MuleSoft*.
. Select *Templates* in the drop-down list to the left of the search bar.

> . To filter the list of templates, add text to the search field. For example, if you enter `Salesforce template`, the search returns all templates that use Salesforce Connector.

Reviewer: ```suggestion
. To filter the list of templates, add text to the search field. 
For example, if you enter `Salesforce template`, the search returns all templates that use Salesforce Connector.
```

. Click on a template to view its details.
+
If a template can be used in Flow Designer, it has an *Open flow* button in the top-right corner of the details page. This option opens the template in read-only mode in Flow Designer, which enables you to find out which connectors and operations are used, how data changes as it moves through the flow, and what configurations for connecting to your systems are already set up.
+
For information about using a downloaded template in Flow Designer, see xref:design-center::import-template.adoc[Create a Mule App Quickly from a Template].
+
All templates can be downloaded to Studio.
+

> . To download a template to Studio, click *Download* on the upper right of the screen. You must be logged into Anypoint Platform to download a template.

Reviewer: To avoid a solitary step, consider something like "All templates...Studio by clicking...screen.

Reviewer: logged into > logged in to
(the action in these cases is "to log in" to rather than "to log into")

+
The system downloads the template as a `.jar` file.

[[import-downloaded-template-studio]]
=== Import a Downloaded Template to Studio

> If you downloaded a template from Exchange, you can import the downloaded `.jar` file into Studio:

Reviewer: ```suggestion
If you downloaded a template from Exchange, you can import the downloaded `.jar` file to Studio:
```

. From Studio, select *Import* from the *File* menu.
. In the Import wizard, expand the *Anypoint Studio* folder.

> . Select *Packaged mule application (.jar)*, and click *Next*.

Reviewer: ```suggestion
. Select *Packaged mule application (.jar)* and click *Next*.
```

. In the *File* field, optionally enter a name for the file.
. Click the ellipses (…​) button.
. Select the `.jar` file to import.
. Click *Open*.
+
Studio imports the template as a new Studio project.

[[import-downloaded-templates-studio]]
=== Add a Template Directly to Studio

When you are working in a Studio, you can import a template directly from Exchange:

> . From Studio, Click the Exchange icon *(X)* in the upper-left of the Studio task bar.

Reviewer: ```suggestion
. From Studio, click the Exchange icon *(X)* in the upper-left of the Studio taskbar.
```

> . Log into Exchange using your Anypoint Platform username and password:

Reviewer: ```suggestion
. Log into Exchange using your Anypoint Platform username and password.
```

> . Search for the template, either directly, or by accessing the template link on the landing page of the associated connector.

Reviewer: ```suggestion
. Search for the template either directly or by accessing the template link on the landing page of the associated connector.
```

. Select the template and click *Open*.
+
Studio imports the template as a new Studio project.

=== Next Step

Follow the instructions on the template's landing page in Exchange to use the template.

== Discover and Add Examples to Studio

> An example is a Studio project that demonstrates how to handle a use case or solution. Many examples include one or more connector configurations. Once you locate an example, you can import it into Studio and then run it.

Reviewer: ```suggestion
An example is a Studio project that demonstrates how to handle a use case or solution. Many examples include one or more connector configurations. After you locate an example, you can import it into Studio and then run it.
```

Discover examples by starting in Exchange or by accessing Exchange from within a Studio project:

> * Start in Exchange to view a list of all, or a subset of MuleSoft-provided examples. From there, you can view download an example to Studio.

Reviewer: ```suggestion
* Start in Exchange to view a list of all or a subset of MuleSoft-provided examples. 
From there, you can view download an example to Studio.
```

* Access Exchange from Flow Designer or Studio to locate an example that fits your project's use case.

When you locate an example in Exchange, the landing page explains how to set up and run the example.

=== Discover All MuleSoft Examples

To use Exchange as a starting point for discovering MuleSoft-provided examples:

. Log into Exchange with your Anypoint username and password.
. Under *All assets* on the left side of the screen, click *Provided by MuleSoft*.
. Select *Examples* in the drop-down list to the left of the search bar.

> . To narrow down the list of examples, enter applicable text in the Search field. For example, if you enter `Salesforce Connector`, the search returns all examples that use the Salesforce Connector.

Reviewer: ```suggestion
. To narrow the list of examples, enter applicable text in the Search field. 
For example, if you enter `Salesforce Connector`, the search returns all examples that use the Salesforce Connector.
```

> . Click on an example to view the example details.

Reviewer: ```suggestion
. Click on an example to view its details.
```

. To download the example for use in Studio, click *Download* on the upper right of the screen.
+
The system downloads the file as a `.jar` file.

=== Import a Downloaded Example to Studio

> If you downloaded an example from Exchange, you can import the downloaded `.jar` file into Studio to create the new project:

Reviewer: ```suggestion
If you downloaded an example from Exchange, you can import the downloaded `.jar` file to Studio to create the new project:
```

. From Studio, select *File > Import*.
. In the Import wizard, expand the *Anypoint Studio* folder.
. Select *Packaged mule application (.jar)*, and click *Next*.
. In the *File* field, optionally enter a name for the file.
. Click the ellipses (…​) button.
. Select the `.jar` file to import.
. Click *Open*.
+
Studio imports the example as a new Studio project.

=== Add an Example Directly to Studio

When you are working in Studio, you can import an example directly from Exchange:

> . From Studio, click the Exchange icon *(X)* in the upper-left of the Studio task bar.:

Reviewer: ```suggestion
. From Studio, click the Exchange icon *(X)* in the upper-left of the Studio taskbar:
```

= Use Exchange to Discover Connectors, Templates, and Examples

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../../assets/images/

> Use Anypoint Exchange (Exchange) to discover connectors, templates, and examples, and to incorporate them into your app. Exchange serves as a repository for connectors, templates, and examples provided both by MuleSoft and by other organizations. Searches in Exchange apply only to the Exchange organization you are viewing.

Reviewer 2: Use Anypoint Exchange to discover integration assets like Connectors, Templates & Examples and import them into your project/app. Exchange is a platform of discover, where API designers, application developers and architects are able to catalog, discover and reuse APIs and integration assets.

You can browse & search for assets in your organization's own Exchange, or you can also browse assets built & curated by MuleSoft in MuleSoft's Exchange, which is publicly available at https://www.mulesoft.com/exchange/, or in your own organization's Exchange, in the left navigation panel from the link: "Provided by MuleSoft". If you are looking for assets in your own organization's Exchange, then you need to have the appropriate permission to view them. 

@wwarner-salesforce : FYI

> Use Anypoint Exchange (Exchange) to discover connectors, templates, and examples, and to incorporate them into your app. Exchange serves as a repository for connectors, templates, and examples provided both by MuleSoft and by other organizations. Searches in Exchange apply only to the Exchange organization you are viewing.

Reviewer 3: "Searches in Exchange apply only to the Exchange organization you are viewing." I think this is incorrect, because if you click "All assets" on the left in Exchange and run a search such as "amazon", it returns both search results from assets in the current Exchange organization and search results from assets in MuleSoft's Exchange (which are shown under the link "Provided by MuleSoft"). I recommend changing those two paragraphs to this:

Use Anypoint Exchange (Exchange) to discover connectors, templates, and examples, and to incorporate them into your app. Exchange serves as a repository for connectors, templates, and examples provided both by MuleSoft and by other organizations. Exchange enables both searching for assets in the Exchange organization you are viewing, and searching MuleSoft's public assets in the https://www.mulesoft.com/exchange/[Public Exchange].

Author: I had updated this to the following. I am sorry that this didn't show up when you looked at it:

Use Anypoint Exchange to discover integration assets such as connectors, templates, and examples and import them into your app or project. You can browse and search for assets built and curated by MuleSoft or assets available in your own organization's exchange.

To access assets in MuleSoft's Exchange, do either of the following:

* Go to https://www.mulesoft.com/exchange/.
* Go to your organization's Exchange and click the *Provided by MuleSoft* link in the left navigation panel.

To access assets in your own organization's Exchange, you must have the appropriate permission.

For more information about searching for assets in Exchange, see xref:exchange::to-find-info.adoc[Find Asset Information].

Author: I would change the second to last paragraph to be this: "To access your organization's Exchange, you must have access rights to that Exchange."

Reviewer 3: Yes, this looks good! I slightly prefer the current version, "you must have the appropriate permission", but the other version sounds fine too.

> Search in the https://www.mulesoft.com/exchange/[Public Exchange] to access MuleSoft's public assets. If you search in a private organization's Exchange, you need access rights to that organization.

Reviewer 4: "If you search in a private organization's Exchange, you need access rights to that organization." -- This is incorrect in that unless you have access to the private Exchange, you won't even see the organization in Exchange.

Author: This was from Kuntal. Are you sure?

Reviewer 3: It's correct that search in private Exchange organizations is access controlled, but confusing to say it this way to users, since they will never see a private Exchange organization but be blocked from searching in it. I omitted this sentence from my edited version, and I think that's the best way to explain Exchange searching as we introduce it in this page.

. To filter the list of connectors, add text to the search field.
+
For example, if you enter `Salesforce`, the search returns all Salesforce connectors.
+
. Click on connector to view its details.

[[add-connector-to-flow-designer]]
=== Add a Connector to a Flow Designer App

When you are working in a Flow Designer project, you can add a connector directly to the project. You can use Flow Designer only for supported connectors.

. Log into Anypoint Platform.
. Open Design Center and create a new Mule app in Flow Designer.
. In the *Let's Get Started wizard*, click *Go straight to canvas*, in the bottom-left corner.
. Click on the Trigger card.
. Select the connector.
. Continue designing the flow, as described in the User Guide for your connector.

[[import-connector-studio]]
=== Import a Connector into a Studio Project

To import a connector into a Studio project:

. From your Studio project, click the Exchange icon *(X)* in the upper-left of the Studio taskbar.
. Log in to Exchange using your Anypoint Platform username and password.
. Search for the connector.
. Select the connector and click *Add to project*.

> . Follow the prompts to add the connector.

Reviewer 4: Please add the alternate Studio install steps here. Once done, then the material in the doc can be consolidated to this topic.
Alternate steps:
. In Studio, create a Mule project.
. In the Mule Palette view, click *(X) Search in Exchange*.
. In *Add Modules to Project*, type "<connector_name>" in the search field.
. Click this connector's name in *Available modules*.
. Click *Add*.
. Click *Finish*.

Author: Is this the favored approach now?

Author: By the way, the most recent Studio version has "Add Dependencies to Project" instead of "Add Modules to Project."

Discover templates by starting in Exchange or by accessing Exchange from within Flow Designer or Studio:

* Start in Exchange to view a list of all or a subset of MuleSoft-provided templates. From there, you can view a template in Flow Designer or download it to Studio.
* Access Exchange from within Flow Designer or Studio to locate a template that fits your use case.

When you locate a template in Exchange, the landing page of the associated connector contains:

* An overview of what the template does
* Prerequisites for using the template
* Instructions for configuring the template

=== Discover MuleSoft Templates

To use Exchange as a starting point for discovering MuleSoft-provided templates:

. https://www.mulesoft.com/exchange/[Log into Exchange].
. Under *All assets* on the left side of the screen, click *Provided by MuleSoft*.
. Select *Templates* in the drop-down list to the left of the search bar.
. To filter the list of templates, add text to the search field.
+
For example, if you enter `Salesforce template`, the search returns all templates that use Salesforce Connector.
+

> . Click on a template to view its details.

Reviewer 4: Style guide says to not use "click on"

Author: Kim reviewed this already, but I'll take a look.

= Use Anypont Exchange to Discover Connectors, Templates, and Examples

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../../assets/images/

Use Anypoint Exchange to discover integration assets such as connectors, templates, and examples and import them into your app or project. You can browse and search for assets built and curated by MuleSoft or assets available in your own organization's Exchange.

> To access assets in MuleSoft's Exchange, do either of the following:

Reviewer: To avoid making a trademark possessive, I suggest using "MuleSoft" here as a plain adjective:
```suggestion
To access assets in MuleSoft Exchange, do either of the following:
```

Reviewer: I guess we refer to "Exchange" only rather than "Exchange instance"? It sounds odd to me to say "your organization's Exchange," but I have had other writers correct me. Would you please let me know your thoughts about this in a separate Slack message?

Author: Saying "MuleSoft Exchange" sounds odd to me. As for saying "Exchange" rather than "Exchange instance", I don't usually work with Exchange, but that's how I've seen it in the doc.

* Go to https://www.mulesoft.com/exchange/.
* Go to your organization's Exchange and click the *Provided by MuleSoft* link in the left navigation panel.

To access assets in your own organization's Exchange, you must have access rights to that Exchange.

For more information about searching for assets in Exchange, see xref:exchange::to-find-info.adoc[Find Asset Information].

> 

Reviewer: end edit here

. Click on a connector to view its details.
. To download the connector for use in Studio, click *Download* from the right menu. You must be logged into Anypoint Platform to download a connector.
+
The system downloads the connector as a `.jar` file. For instructions on adding this file to Studio, see <<import-downloaded-connector-studio.adoc, Import a Downloaded Connector into Studio>>.

You can also add a connector directly to a Flow Designer or Studio project, as described later in this topic.

[[add-connector-to-flow-designer]]
=== Add a Connector to a Flow Designer App

When you are working in a Flow Designer project, you can add a connector directly to the project:

. Log into Anypoint Platform.
. Open Design Center and create a new Mule app.
. In the *Let's Get Started wizard*, click *Go straight to canvas*, in the bottom-left corner.
. Click the on Trigger card.
. Select the connector.
. Continue designing the flow, as described in the User Guide for your connector.

[[import-downloaded-connector-studio]]
=== Import a Downloaded Connector into a Studio

> If you downloaded a connector from Exchange, you can import the downloaded `.jar` file into a Studio project:

Reviewer 5: AFAIK this is incorrect. There isn't a way how to import connector into the Studio like that, this is only possible for templates, examples/demos, etc. (within open mule application project). @adam-lipka please double check this.
